stdlib/ballerina-mime/src/main/ballerina/ballerina/mime/natives.bal

```diff
@@ -276,3 +276,87 @@ public const string DEFAULT_CHARSET = "UTF-8";
+@Param {value:"entity: Represent the MIME entity"}
+@Param {value:"headerName: Represent header name"}
+@Return {value:"Return header value associated with the given header name. If multiple header values are present,
+then the first value will be returned"}
+public function <Entity entity> getHeader (string headerName) (string) {
+    if (entity.headers == null) {
+              return null;
+    }
+    var headerValue, _ = (string[])entity.headers[headerName];
+    return headerValue == null ? null : headerValue[0];
+}
+
+@Description {value:"Get all the header values associated with the given header name"}
+@Param {value:"entity: Represent a MIME entity"}
+@Param {value:"headerName: Represent the header name"}
+@Return {value:"Return all the header values associated with the given header name as a string of arrays"}
+public function <Entity entity> getHeaders (string headerName) (string[]) {
+    if (entity.headers == null) {
+        return null;
+    }
+    var headerValue, _ = (string[]) entity.headers[headerName];
+    return headerValue;
+}
+
+@Description {value:"Get all the headers"}
+@Param {value:"entity: Represent a MIME entity"}
+@Return {value:"Return all the headers in the entity"}
+public function <Entity entity> getAllHeader () (map){
+    if (entity.headers == null) {
+        return null;
+    }
+    return entity.headers;
+}
+
+@Description {value:"Add the given header value against the given header"}
+@Param {value:"entity: Represent a MIME entity"}
+@Param {value:"headerName: Represent the header name"}
+@Param {value:"headerValue: Represent the header value to be added"}
+public function <Entity entity> addHeader (string headerName, string headerValue){
+    if (entity.headers == null) {
+        entity.headers = {};
+    }
+    var existingValues = entity.headers[headerName];
+    if (existingValues == null) {
+        entity.setHeader(headerName, headerValue);
+    } else {
+        var valueArray, _ = (string[]) existingValues;
+        valueArray[lengthof valueArray] = headerValue;
+        entity.headers[headerName] = valueArray;
+    }
+}
+
+@Description {value:"Set the given header value against the given header. If a header already exist, its value will be
+replaced with the given header value"}
+@Param {value:"entity: Represent a MIME entity"}
+@Param {value:"headerName: Represent the header name"}
+@Param {value:"headerValue: Represent the header value"}
+public function <Entity entity> setHeader (string headerName, string headerValue) {
+    if (entity.headers == null) {
+        entity.headers = {};
+    }
+    string[] valueArray = [headerValue];
+    entity.headers[headerName] = valueArray;
+}
+
+@Description {value:"Remove the given header from the entity"}
+@Param {value:"entity: Represent a MIME entity"}
+@Param {value:"headerName: Represent the header name"}
+public function <Entity entity> removeHeader (string headerName) {
+    if (entity.headers != null) {
+        entity.headers.remove(headerName);
```

Comment:
If entity.headers represents a map, map remove will return a boolean with recent changes. In that case you may need to ignore it. 
_ = entity.headers.remove(headerName);
Can you please check this PR against the new master changes?

Reply:
We will be removing the map from entity struct, so will update the PR accordingly. Thank you.